control-plane/api-gateway/consul/config_entry_translation.go

```diff
@@ -82,8 +90,198 @@ func (t Translator) GatewayToAPIGateway(k8sGW gwv1beta1.Gateway, certs map[types
 		},
 		Listeners: listeners,
 		Partition: consulPartition,
-		Namespace: namespaces.ConsulNamespace(k8sGW.GetObjectMeta().GetNamespace(), t.EnableK8sMirroring, t.ConsulDestNamespace, t.EnableK8sMirroring, t.MirroringPrefix),
+		Namespace: t.getConsulNamespace(k8sGW.GetObjectMeta().GetNamespace()),
+	}
+}
+
+// HTTPRouteToHTTPRoute translates a k8s http route into a consul config http route.
+func (t Translator) HTTPRouteToHTTPRoute(k8sHTTPRoute gwv1beta1.HTTPRoute, parentRefs map[types.NamespacedName]consulIdentifier) capi.HTTPRouteConfigEntry {
+	consulPartition := os.Getenv("CONSUL_PARTITION")
+
+	routeName := k8sHTTPRoute.Name
+	if routeNameFromAnnotation, ok := k8sHTTPRoute.Annotations[AnnotationHTTPRoute]; ok && routeNameFromAnnotation != "" && !strings.Contains(routeNameFromAnnotation, ",") {
+		routeName = routeNameFromAnnotation
+	}
+
+	consulHTTPRoute := capi.HTTPRouteConfigEntry{
+		Kind: capi.HTTPRoute,
+		Name: routeName,
+		Meta: map[string]string{
+			metaKeyManagedBy:       metaValueManagedBy,
+			metaKeyKubeNS:          k8sHTTPRoute.GetObjectMeta().GetNamespace(),
+			metaKeyKubeServiceName: k8sHTTPRoute.GetObjectMeta().GetName(),
+		},
+		Partition: consulPartition,
+
+		Namespace: t.getConsulNamespace(k8sHTTPRoute.GetObjectMeta().GetNamespace()),
+	}
+
+	// translate hostnames
+	hostnames := make([]string, 0, len(k8sHTTPRoute.Spec.Hostnames))
+	for _, k8Hostname := range k8sHTTPRoute.Spec.Hostnames {
+		hostnames = append(hostnames, string(k8Hostname))
+	}
+	consulHTTPRoute.Hostnames = hostnames
+
+	// translate parent refs
+	consulHTTPRoute.Parents = translateHTTPRouteParentRefs(k8sHTTPRoute.Spec.CommonRouteSpec.ParentRefs, parentRefs)
+
+	// translate rules
+	consulHTTPRoute.Rules = t.translateHTTPRouteRules(k8sHTTPRoute.Spec.Rules)
+
+	return consulHTTPRoute
+}
+
+func translateHTTPRouteParentRefs(k8sParentRefs []gwv1beta1.ParentReference, parentRefs map[types.NamespacedName]consulIdentifier) []capi.ResourceReference {
+	parents := make([]capi.ResourceReference, 0, len(k8sParentRefs))
+	for _, k8sParentRef := range k8sParentRefs {
+		parentRef, ok := parentRefs[types.NamespacedName{Name: string(k8sParentRef.Name), Namespace: string(*k8sParentRef.Namespace)}]
+		if !ok {
+			// we drop any parent refs that consul does not know about
+			continue
+		}
+		ref := capi.ResourceReference{
+			Kind:        capi.APIGateway, // Will this ever not be a gateway? is that something we need to handle?
```

Comment:
can we rely on a parent ref always being a gateway?

Reply:
no, we can't, we'll need to revalidate here and only translate refs that correspond to gateways since that's all we can handle (but the spec allows for refs of other implementation-defined types too), but I'm thinking that maybe we can just add this as a continuation clause a bit earlier? Something like this logic (which will also likely need to exist in the controller itself, so we'll likely want to eventually extract into something like ahelper somewhere):

```
if (k8sParentRef.Kind != nil && string(*k8sParentRef.Kind) != apiGatewaySpecKind) || 
   (k8sParentRef.Group != nil && string(*k8sParentRef.Group) != apiGatewaySpecGroup) {
  continue
} 
```